marginalia.el

```diff
@@ -730,16 +762,47 @@ These annotations are skipped for remote paths."
       (marginalia--fields ("*Remote*" :face 'marginalia-documentation))
     (when-let (attributes (file-attributes (substitute-in-file-name (marginalia--full-candidate cand)) 'string))
       (marginalia--fields
-       ((file-attribute-modes attributes) :face 'marginalia-file-modes)
-       ((format "%s:%s"
-                (file-attribute-user-id attributes)
-                (file-attribute-group-id attributes))
+       ((concat (unless (string= user-login-name (file-attribute-user-id attributes))
+                  (file-attribute-user-id attributes))
+                (let ((group (file-attribute-group-id attributes)))
+                  (if (numberp group)
+                      (unless (= (group-gid) group)
+                        (concat ":" (number-to-string group)))
+                    (unless (string= (group-name (group-gid)) group)
+                      (concat ":" group)))))
         :width 12 :face 'marginalia-file-owner)
+       ((marginalia--color-file-attributes (file-attribute-modes attributes)))
        ((file-size-human-readable (file-attribute-size attributes)) :width 7 :face 'marginalia-size)
        ((format-time-string
          "%b %d %H:%M"
          (file-attribute-modification-time attributes)) :face 'marginalia-date)))))
 
+(defvar marginalia--color-file-attributes-cache nil
```

Comment:
hashtable?

Reply:
I tried a hashtable and an alist, the alist was actually slightly faster.